### ISSUE TITLE
[REVIEW]: WDPM: the Wetland DEM Ponding Model

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2276]
**Submitting author:** @KevinShook (<a href="http://orcid.org/0000-0003-0363-9929">Kevin Shook</a>)
**Repository:** <a href="https://github.com/CentreForHydrology/WDPM" target ="_blank">https://github.com/CentreForHydrology/WDPM</a>
**Version:** v2.0b
**Editor:** @kbarnhart
**Reviewer:** @r-barnes, @awickert, @KCallaghan
**Archive:** <a href="https://doi.org/10.5281/zenodo.5165172" target="_blank">10.5281/zenodo.5165172</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/407a1b519d6242721f056cfeee7669ec/status.svg)](https://joss.theoj.org/papers/407a1b519d6242721f056cfeee7669ec)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/407a1b519d6242721f056cfeee7669ec"><img src="https://joss.theoj.org/papers/407a1b519d6242721f056cfeee7669ec/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/407a1b519d6242721f056cfeee7669ec/status.svg)](https://joss.theoj.org/papers/407a1b519d6242721f056cfeee7669ec)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/CentreForHydrology/WDPM" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@r-barnes & @awickert & @KCallaghan, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @kbarnhart know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/CentreForHydrology/WDPM">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@KevinShook) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/CentreForHydrology/WDPM">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@KevinShook) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/CentreForHydrology/WDPM">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@KevinShook) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @r-barnes, @awickert, @KCallaghan it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1002/hyp.6787 is OK
- 10.1002/hyp.8381 is OK
- 10.1002/hyp.9867 is OK
- 10.1016/j.jhydrol.2014.12.014 is OK
- 10.1016/S0022-1694(03)00175-6 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02276/joss.02276/10.21105.joss.02276.pdf)

**Comment by kbarnhart** [comment]
@KevinShook, @r-barnes, @awickert, @KCallaghan

I wanted to check in and see how things were going with this review, and to answer any questions if you have them. As mentioned above, JOSS is presently requesting reviewers complete their reviews in six weeks (a bit more than three weeks from now). 

@KCallaghan thanks for linking all of your issues to this one 👍 🎉

**Comment by KevinShook** [comment]
There are still some outstanding issues with the program, particularly in terms of the installation.
This may be because the code has not been looked at for over 6 years.
We are attempting to fix the issues.

**Comment by awickert** [comment]
@kbarnhart The review is in the schedule! It should reach the top of the queue in 1.5 weeks.

**Comment by kbarnhart** [comment]
All- Thanks for the updates. Please let me know if I can do anything to support the review process.

**Comment by KevinShook** [comment]
Thanks very much - will do

**Comment by r-barnes** [comment]
That's it for me this morning; I'll circle back to inspecting the C and OpenCL code later.

**Comment by kbarnhart** [comment]
Thanks everyone for linking issues and keeping everyone updated on this issue thread. As always, authors and reviewers, please feel free to reach out to me for support in this review process.

**Comment by awickert** [comment]
@kbarnhart - I didn't accept the invite in time and it expired, so I cannot update the checklist. Can you re-send? (With apologies!)

I finished a first pass through, and will be able to add more material once I am able to compile the source code.

**Comment by danielskatz** [comment]
@whedon re-invite @awickert as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@awickert please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by kbarnhart** [comment]
Thanks @danielskatz !

**Comment by kbarnhart** [comment]
@KevinShook and @r-barnes @awickert @KCallaghan 

I wanted to write a quick message to check in and provide some guidance regarding the JOSS process. As best as I can tell the reviewers have been quite thorough and have opened many issues in the source repository as part of their reviews. 

At this point, the next step is for the authors to address these issues and close them when completed. If the authors have clarifying questions for the reviewers or for me (e.g., does this pull request address your concern), please tag us into the conversation. If the authors think that an issue that was opened is out of scope, the issue itself is the place to discuss. As always, please consider me a resource throughout this process. 

Once the authors feel like they have addressed the reviewers issues such that the reviewers are able to complete the review checklist, they should comment here indicating this. 

Know that I'll check into this thread and the main repository to keep track of progress/status changes at a frequency of about every 1-2 weeks. 

Questions, comments, or concerns always welcome.

**Comment by KevinShook** [comment]
Thanks very much, this is very helpful for understanding the process.
We'll work on addressing the issues raised.

**Comment by kbarnhart** [comment]
@KevinShook I wanted to check in and see how things were going with addressing the issues raised by reviewers. Know that JOSS doesn't have a required timeline for revisions, but it would be helpful for me to know an approximate time frame.

**Comment by KevinShook** [comment]
Thanks very much for keeping in touch.
We have almost completed the conversion of the GUI from Python 2 to 3.
Once the tested is complete then the rest of the issues should be fairly simple to address.
I have started editing the paper itself, and the requested clarifications seem to be relatively easily addressed.
Barring unforeseen issues, I would anticipate completion sometime this month.

**Comment by kbarnhart** [comment]
@KevinShook Thanks for the update. If issues come up that I can assist with, please let me know.

**Comment by kbarnhart** [comment]
@whedon remind kbarnhart in 2 weeks

@KevinShook just setting a reminder for myself to check in on this in a few weeks.

**Comment by whedon** [comment]
Reminder set for kbarnhart in 2 weeks

**Comment by kbarnhart** [comment]
@KevinShook I wanted to check in and see how things were going with addressing reviewer comments.

**Comment by KevinShook** [comment]
Thanks for getting in touch.
I just merged a pull request today which should have all of the fixes to date.
The GUI has been converted from Python 2 to 3, and no-longer crashes.
I believe that the style issues have also been addressed.
The main outstanding issues are a) there's still one warning left to fix, and b) although we have a makefile, it's not included as it needs a bit of work.
The paper has been updated, but still needs to be gone over, and the manual is being updated as well.

**Comment by r-barnes** [comment]
@KevinShook: I'd recommend `cmake` as a better option than a makefile ([example here](https://github.com/r-barnes/Barnes2020-FillSpillMerge/blob/master/CMakeLists.txt)). If you'd like help getting one setup, I'm happy to help.

**Comment by KevinShook** [comment]
Thanks very much. We are planning on going with cmake as was recommended by one of the developers.

**Comment by kbarnhart** [comment]
Thanks for the update @KevinShook 👍

**Comment by kbarnhart** [comment]
@KevinShook another quick check in from me. No rush if you and your co-authors are still addressing issues raised by the reviewers.

**Comment by KevinShook** [comment]
Thanks for checking in. 
We are still finishing a few issues.
We are replacing the makefile with cmake as was suggested by r-barnes (thanks!).
We are also working on having the program work with a single executable. Previously, the GUI called a separate executable, from the command line version, which is obviously not desireable.
One these are done, the manual will need to be updated.
I'm going away for 3 weeks, so will probably not be able to work on it much in the interim.

**Comment by kbarnhart** [comment]
@KevinShook Sounds good, thanks for the update.

**Comment by kbarnhart** [comment]
@KevinShook another quick check in from me. No rush if you and your co-authors are still addressing issues raised by the reviewers.

**Comment by KevinShook** [comment]
Thanks. We have finally solved a problem which was responsible for there being 2 different executables. There is now a single executable, but we are working on making it more efficient. When this last hurdle is cleared, we can finally cross off all of the technical issues, and finish the documentation.

**Comment by kbarnhart** [comment]
@KevinShook another quick check in from me (I'm trying to do this about once a month). Of course no rush if you and co-authors are still working to address the remaining technical and documentation issues.

**Comment by KevinShook** [comment]
Thanks very much.
We're nearly finished. The revised code has been pushed to our GitHub repo, as has the revised paper.
I believe that we have addressed all of the concerns with the program, and
I'm just waiting for the approval of two the co-authors on the paper and the program manual,
to reply to all of the issues.

**Comment by kbarnhart** [comment]
@KevinShook thanks for that update. I'll stay tuned for your response to the in-repo issues.

**Comment by kbarnhart** [comment]
@KevinShook I wanted to touch base again on how finishing addressing the reviewer issues was going.

**Comment by KevinShook** [comment]
Thanks for keeping in touch.
I have replied to all but 3 issues, and I believe that all but these last 3 have been solved.
I didn't close the issues, as I assumed that this would be up to the person creating them.

The program now runs (much better thanks to the reviewers), and the GUI has also been greatly
improved.
The last 3 issues (#18, #14, and #13) are concerned with programming style and are being addressed right now.
They should be finished in the next few days.

**Comment by kbarnhart** [comment]
@KevinShook Thanks for this update. Glad to know that the review comments have improved the software. I concur with you that the reviewers who created the issues should be the people who close them. 

Once you have finished addressing the final issues, please ping me and the reviewers here to let them know that it is time for them to complete their reviews.

**Comment by kbarnhart** [comment]
@KevinShook another quick check in from me to see how things are going in finishing addressing the reviewer's comments.

**Comment by KevinShook** [comment]
I think that we have addressed all of the comments. I'm just waiting for the final sign-off from the last co-author before contacting you and the other reviewers. I did see that some of the comments have already been closed.

**Comment by kbarnhart** [comment]
@KevinShook thanks for the update!

**Comment by KevinShook** [comment]
The paper is finally complete and has been pushed to the repo.
Thanks for your patience!
What would be the best way for me to contact all the reviewers to tell them
that we have finished?

**Comment by kbarnhart** [comment]
@whedon generate pdf

**Comment by kbarnhart** [comment]
@whedon check references

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02276/joss.02276/10.21105.joss.02276.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02276/joss.02276/10.21105.joss.02276.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1029/2020WR027984 is OK
- 10.1002/hyp.10567 is OK
- 10.1007/s10669-018-9684-7 is OK
- 10.1111/j.1745-6584.1988.tb00794.x is OK
- 10.1002/esp.3888 is OK
- 10.1002/hyp.6787 is OK
- 10.1002/hyp.8381 is OK
- 10.1002/hyp.9867 is OK
- 10.1016/j.jhydrol.2014.12.014 is OK
- 10.1016/S0022-1694(03)00175-6 is OK

MISSING DOIs

- 10.1016/s0022-1694(98)00098-5 may be a valid DOI for title: Water and Solute Transfer between a Prairie Wetland and Adjacent Uplands, 1. Water Balance
- 10.3390/w11122486 may be a valid DOI for title: “Garbage in, Garbage Out” Does Not Hold True for Indigenous Community Flood Extent Modeling in the Prairie Pothole Region
- 10.1016/0022-1694(93)90275-e may be a valid DOI for title: Hydrology of a Prairie Slough

INVALID DOIs

- None
```

**Comment by kbarnhart** [comment]
@KevinShook thanks for the update. 

@r-barnes @awickert @KCallaghan as you are able, please revisit your review at this point. If you think @KevinShook has addressed all of your concerns, please close in-repo issues and check off remaining checkboxes above.

As always, if you have questions for me ask them here or email krbarnhart@usgs.gov

**Comment by KevinShook** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1029/2020WR027984 is OK
- 10.1002/hyp.10567 is OK
- 10.1007/s10669-018-9684-7 is OK
- /10.1016/S0022-1694(98)00098-5 is OK
- 10.1111/j.1745-6584.1988.tb00794.x is OK
- 10.1002/esp.3888 is OK
- 10.1002/hyp.6787 is OK
- 10.1002/hyp.8381 is OK
- 10.1002/hyp.9867 is OK
- 10.1016/j.jhydrol.2014.12.014 is OK
- 10.1016/S0022-1694(03)00175-6 is OK
- 10.3390/w11122486 is OK
- 10.1016/0022-1694(93)90275-E is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by KevinShook** [comment]
I added the missing DOIs

**Comment by KevinShook** [comment]
A strange (but welcome) thing just happened. Out of the blue, Oluwaseun Sharomi contacted us!
It turns out he had moved from Abu Dhabi to the USA, which was why we were unable to contact him.
He has given his consent to be a co-author, so I have added him to the paper.

**Comment by awickert** [comment]
Hi @KevinShook. First off, fantastic news about Oluwaseun Sharomi. This is really an ideal solution. Second, once you sent your message above about completion, I began to tick off the remaining items. One that seemed a sticky point that I should have addressed earlier is "State of the field: Do the authors describe how this software compares to other commonly-used packages?". I have not seen much on this; it might be good to include a paragraph or so. (If I've missed something in my new skim, please just note this.)

Otherwise, I simply need to complete my own tests using it, which I hope to do by the end of next week (sorry for the bit of time, my schedule is packed). So far, so good; hoping that I'll be able to simply tick the boxes, but doing the due diligence.

**Comment by KevinShook** [comment]
HI @awickert 
We're very relieved to be able to include Oluwaseun Sharomi. As you said, this is by far the best solution. 

As for your point, we did describe the differences between WDPM and other types of programs which are used in the region.

We first explained that WDPM is not a hydrological model:

> WDPM does not attempt to compute the magnitudes of the additive or subtractive fluxes, which must be established by measurement or through the use of a physically based model of prairie hydrology such as the Cold Regions Hydrological Modelling platform (CRHM) [@pomeroyColdRegionsHydrological2007].

We then explained the differences between WDPM and hydraulic models, in that WDPM is a model of the state of the
landscape after the fluxes have completed:

> The  model output is the depth of water over each cell of the DEM. WDPM is not a hydraulic model; it is a model of the state of the distribution of water in a landscape after the fluxes have been redistributed. FLUXOS-OVERFLOW is a 2-D hydrodynamic model capable of modelling overland flow rates in the prairies [@costaPredictingVariableContributing2020], but it is much slower than WDPM, and it requires more parameters and more forcing data.

I'm not sure if this is sufficient, or we need to add some further explanations.

**Comment by kbarnhart** [comment]
@KevinShook great news about Oluwaseun Sharomi. Thanks for that update. 

@awickert thanks for the update regarding revisiting your review.

**Comment by awickert** [comment]
Hi @KevinShook. Regarding your background, you do indeed compare your work to different hydrological models. However, there are a number of computational tools that are quite a lot more similar to what you are proposing here. Some references pertaining to the Canadian prairies include:
* Li, S., MacMillan, R., Lobb, D. A., McConkey, B. G., Moulin, A., and Fraser, W. R.: Lidar DEM error analyses and topographic depression identification in a hummocky landscape in the prairie region of Canada, Geomorphology, 129, 263–275, https://doi.org/10.1016/j.geomorph.2011.02.020, 2011.
* Shaw, D. A., Pietroniro, A., and Martz, L. W.: Topographic analysis for the prairie pothole region of Western Canada, Hydrol. Process., 27, 3105–3114, https://doi.org/10.1002/hyp.9409, 2013.

The Shaw reference seems quite close to your work here.

In addition, while ethically I can by no means suggest that you cite work by your referees here, I would like to point you towards the set of three papers that we have recently written/published; the first one (FlowFill) seems very similar to your approach here. The second and third address the issues of computational efficiency using a graph. Perhaps they would be of more general interest?
* Callaghan, K. L. and Wickert, A. D.: Computing water flow through complex landscapes – Part 1: Incorporating depressions in flow routing using FlowFill, Earth Surf. Dynam., 7, 737–753, https://doi.org/10.5194/esurf-7-737-2019, 2019.
* Barnes, R., Callaghan, K. L., and Wickert, A. D.: Computing water flow through complex landscapes – Part 2: Finding hierarchies in depressions and morphological segmentations, Earth Surf. Dynam., 8, 431–445, https://doi.org/10.5194/esurf-8-431-2020, 2020.
* Barnes, R., Callaghan, K. L., and Wickert, A. D.: Computing water flow through complex landscapes – Part 3: Fill–Spill–Merge: flow routing in depression hierarchies, Earth Surf. Dynam., 9, 105–121, https://doi.org/10.5194/esurf-9-105-2021, 2021.

In addition, our above papers spend some time discussing past work in managing flow through depressions -- indeed, this is where I found the references that I note above.

I hope that this is helpful! Again, and to be explicit: this is not a request to cite the referees' work! But some acknowledgment of prior research in the same vein would, I think, be necessary background for your paper.

**Comment by KevinShook** [comment]
Thanks for the suggestions. I was trying to keep the paper to be as short as possible, but will elaborate on other models
WDPM is very different from the work of Shaw et al, which is based on the modelling of simulated individual depressions, rather than a DEM - we have expanded on this methodology in another one of our models, which I was trying to avoid discussing.

WDPM is also unlike programs like FlowMapR, used by Li et al,, and other similar GIS programs, which quantify _filled_ depressions - the intent of WDPM is to quantity the states of water in DEMs which are _not_ completely filled, and to demonstrate the difference in landscape connectivity between filling and emptying fluxes.

The three papers that you have published look to be very interesting, and do seem to be much more similar to our methodology. I am sorry to admit that I was not aware of their existence. I will read them over very carefully.

**Comment by awickert** [comment]
In this case, don't feel any need to discuss unrelated literature -- I think that it is simply that I saw that your background included hydrologic and hydraulic models, but I did not see explicit references to prior approaches to manage depressions, which seems to be a key methodological issue that you solve.

I believe that our "FlowFill" is largely the same as your apprpoach, and that our DH+FSM may solve for the desired fields much more efficiently. This is not to detract from the value of WDPM, but is to say that we could perhaps follow up after this review to see what might be helpful in terms of computational efficiency.

**Comment by KevinShook** [comment]
Thanks. I'm going to add a brief section differentiating WDPM from other models, which may avoid some confusion.
I'd be very interested in following up offline. WDPM is not very computationally efficient, but it does
have some advantages over stateless models, as its purpose was to model the states of depressional storage
(the spatial distributions of water, and the connected/contributing fractions), under repeated cycles of additions
and removals of water, which few other models do.

**Comment by KevinShook** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.envsoft.2020.104850 is OK
- 10.1016/j.envsoft.2020.104850 is OK
- 10.5194/esurf-8-431-2020 is OK
- 10.5194/esurf-9-105-2021 is OK
- 10.5194/esurf-7-737-2019 is OK
- 10.1029/2020WR027984 is OK
- 10.1002/hyp.10567 is OK
- 10.1007/s10669-018-9684-7 is OK
- /10.1016/S0022-1694(98)00098-5 is OK
- 10.1111/j.1745-6584.1988.tb00794.x is OK
- 10.1002/esp.3888 is OK
- 10.1002/hyp.6787 is OK
- 10.1002/hyp.9409 is OK
- 10.1002/hyp.8381 is OK
- 10.1016/j.jhydrol.2020.125846 is OK
- 10.1002/hyp.9867 is OK
- 10.1016/j.jhydrol.2014.12.014 is OK
- 10.1016/S0022-1694(03)00175-6 is OK
- 10.3390/w11122486 is OK
- 10.1016/0022-1694(93)90275-E is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by KevinShook** [comment]
Have added the additional comparisons as discussed with @awickert.
BTW, you were right about Shaw et al - I was thinking of another one of his models.
I've also added a reference to the PRIMA model.

**Comment by awickert** [comment]
Thanks, Kevin! I was starting to doubt my memory of that article. I'm also glad that you note the importance of WDPM simulating water removal. (Actually, @KCallaghan, @r-barnes, and I are wrapping up a manuscript that includes evaporation as well as groundwater... let's talk about this when we connect offline too.)
I'll tick the final box after I finish testing your model on my computer. I'll try to be semi-quick about this!

**Comment by KevinShook** [comment]
That's great - I'm looking forward to talking about this.
BTW, I was running WDPM w/o the GUI today and found that the switch to select the CPU/GPU for parallel
processing had been reversed - it should be 0 for the CPU, and 1 for the GPU, but the program was using the reverse. It doesn't affect the GUI, but it's being fixed right now.

**Comment by kbarnhart** [comment]
@awickert thanks for these constructive comments associated with revisiting your review. 

@KevinShook thanks for being so responsive regarding these additional comments and the updates regarding DOIs and the CPU flag change. 

@r-barnes and @KCallaghan as you are able, please revisit your reviews and provide an update here.

**Comment by KevinShook** [comment]
The CPU/GPU switch has been fixed so that it now corresponds to the documentation.

**Comment by kbarnhart** [comment]
@r-barnes and @KCallaghan a friendly reminder to revisit your review of this contribution now that the author has worked to address issues raised.

**Comment by kbarnhart** [comment]
FYI, I will be out of the office from April 15 until April 20. The ooo bot should respond indicating this if you tag me during that time. I will respond to any comments once I return.

**Comment by r-barnes** [comment]
I hope to have time this weekend or late Friday night :-(

**Comment by r-barnes** [comment]
I was able to make another pass today. 

Some issues:
* The build system seems complicated versus what's needed. I've put up a PR CentreForHydrology/WDPM#56 that shows how I'd do this. However, I have not been able to test it on Mac/Windows.
* I've made a few other comments regarding code quality.

I think my biggest concern is that so much of the documentation is wrapped up in a PDF file. I don't really see the advantage of this. In particular, I opened the repo again and was immediately lost as to how to install/use the software. It took me 5 clicks and a Ctrl+F to begin to track down this information. Including this information in the README or linking to explanatory markdown seems like a better approach to me. I'm not sure if JOSS has a policy one way or the other on this.

**Comment by kbarnhart** [comment]
Thanks for those comments @r-barnes. I'm just back from being out of office. 

Wanted to acknowledge that I'll weigh in on the comment above about docs, but can't get to it today. Should be by tomorrow or Friday.

**Comment by awickert** [comment]
I have finished my tests of WDPM (it worked) and ticked off the remaining items in my review.

**Comment by kbarnhart** [comment]
@awickert thanks for that update and for completing your review! 🎉 

@KCallaghan any chance you could provide an update on your availability to complete your review (note, have also emailed). 

Finally, regarding @r-barnes question regarding the status of the documentation in a PDF (vs some other format). Thanks for your patience on this, and I see that @KevinShook has made some updates to the Readme.md in the interim. 

JOSS doesn't have a policy regarding the format/hosting of the documentation. This comes in part from wanting to provide flexibility to authors in how to handle documentation that meets the criteria listed in the "documentation" section, while respecting the diversity of software types and languages we see. I've handled examples that span from a docx and pdf, to a GitHub Wiki, Jupyter Notebooks, to a sphinx/doxogen approach. Some examples use a hybrid where the documentation may span multiple mediums. When a pdf is used it is definitely a good practice to provide the source (e.g., in this case the lyx file used and the figures, or in other cases the source docx). 

I think the key considerations here are:

1. Is it immediately apparent where the documentation elements are located? This may mean pointing the potential user towards one or more resources if documentation elements are located in multiple locations. Documentation should be obtainable without login/registration ([as per submission requirements](https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements)).

2. Does the documentation contain the required information (statement of need, installation instructions, example usage, functionality documentation, automated tests or similar, community guidelines)?

3. Are each of these elements clear and usable (e.g., a non searchable PDF would be unusable)?

Let me know if there is other clarification I can provide.

**Comment by KCallaghan** [comment]
@kbarnhart I apologise for being the weak link in this review! This is my top priority to work on next week.

**Comment by kbarnhart** [comment]
No worries @KCallaghan. Thanks for the update and look forward to seeing your review. Have a good weekend!

**Comment by KCallaghan** [comment]
It looks like the authors have handled all of the issues I brought up previously, thank you! I found running and testing the software significantly easier this time around. I feel comfortable ticking off the remaining items in my review. I do have one note:
 - Page 26 of the manual details the fields to enter when running from the command line, but leaves out the 'water runoff fraction', which can lead to confusion (the error message that comes up in the command line shows the correct order of fields)
 
 And one question:
  - Is there a particular use case in which you think it would be sensible to 'add', and _not_ 'drain'? It seems as though the examples you give require add then drain, and I was wondering why the algorithm initially dams water at the edges of the DEM during the add step rather than allowing water to flow out of the DEM during this step (essentially combining 'add' and 'drain' into one).

**Comment by KevinShook** [comment]
Whoops! Thanks for finding the error on P. 26. I'll add it right away.

There are a couple of reasons why you might not want to drain.
One is that in many prairie basins there is no need to drain, as the basin is closed (does not spill)
for any foreseeable addition of water. There's no point in draining these basins; in fact draining
can cause problems, because the program will be trying to remove water from the lowest point,
which would be at the bottom of a depression as the "outlet" lies above much of the basin.

The figure shows one of these basins, where we do a lot of work, near Saskatoon. The figure the effect of adding nearly 2 years of precipitation with 100% runoff and zero evaporation/infiltration, which would be required to fill the basin. 
Unsurprisingly, in 50+ years of study, the basin has never spilled. My standard joke is that the last time it spilled was when the glaciers melted!

![600_0_0_0_u](https://user-images.githubusercontent.com/11050573/116308162-6d76c300-a764-11eb-84f3-2ed741393952.png)

Another reason for draining separately is that it might be desirable to use differing precisions when adding/removing water
and draining. When several additions and/or removals are specified as a sequence, the water areas
are required to be accurate, but the water depths are not as important. Therefore, a fairly coarse 
tolerance can be used. When draining a basin (particularly for estimating the contributing area),
finer tolerances are usually desirable.

**Comment by KCallaghan** [comment]
Thanks for clarifying!

**Comment by KevinShook** [comment]
Glad I could make it clearer. It's such a weird area.
I updated the manual in commit https://github.com/CentreForHydrology/WDPM/commit/fab593467c62d921c92fc16556add57c5a23b0b5

**Comment by kbarnhart** [comment]
@KCallaghan thanks for completing your review and finding that issue. @KevinShook thanks for addressing these issues quickly as they arise and providing the above example. Very interesting. 

@r-barnes as you are able please indicate if there are any remaining concerns identified by your review. As always, let me know if I can assist or clarify in the process.

**Comment by kbarnhart** [comment]
@r-barnes a friendly reminder to revisit your review. 

Also a comment to all. I will be in the field all of next week, so will be delayed in my response during that time. I'll address any comments the week of 5/17.

**Comment by r-barnes** [comment]
Thanks @kbarnhart, I'm in the process of looking through things now.

**Comment by kbarnhart** [comment]
@r-barnes 👍 thanks for the update and for taking the time to revisit this review.

**Comment by r-barnes** [comment]
Okay, I've made another review. There are a few things left to fix.

* [ ] **Tests.** The software doesn't seem to have tests associated with it. The manual obliquely refers to a sample `basin5.asc` file which could be processed; however, this would only show that the software does something. Automating this simple test of functionality with unittest+pytest should be pretty straight-forward. However, more extensive tests are desirable. For instance, my recent FillSpillMerge work performs a very similar operation to what the authors describe and during testing we discovered several subtle bugs. After this recent scarring experience I've been trying to more thoroughly test all my own code! In FillSpillMerge I used randomized testing to generate hundreds of thousands of artificial landscapes and then tested (1) to see if adding a very large amount of water gives a result exactly equal to depression filling, (2) that incrementally adding portions of water gives the same result as adding the same amount of water all at once, and (3) more tests as described [here](https://github.com/r-barnes/Barnes2020-FillSpillMerge/blob/master/unittests/fsm_tests.cpp). I'm glad that my tests allowed me to identify and fix bugs in my code before they impacted downstream science. Given the complexity of WDPM having a good set of unittests seems appropriate. CentreForHydrology/WDPM#67
* [ ] **Contributing.** I couldn't find a clear contributor guide. I expected one in a `CONTRIBUTING.md` file but don't see that or "Contributing" in the manual. Regardless of whether this material is present in the manual, I think it should be in a `CONTRIBUTING.md` file because this is, logically, where a contributor would look for it. CentreForHydrology/WDPM#88
* [ ] **Functionality documentation.** Certain GUI elements seem to be undocumented. Using Tooltips to provide documentation information in-program seems like a reasonable solution. CentreForHydrology/WDPM#69
* [ ] **Functionality.** As above, I can confirm that the software does something. However, I cannot confirm that it does exactly what it claims it does. A test suite such as that discussed above would make a strong argument in favour of functionality.
* [ ] **State of the Field**. My FillSpillMerge software, mentioned above and described in [two](https://esurf.copernicus.org/articles/8/431/2020/) [papers](https://esurf.copernicus.org/articles/9/105/2021/), seems to perform a very similar operation to what the authors describe here. Admittedly, I have a bias/conflict of interest here, so I'm not interested in pressing this issue if the authors prefer not to mention it.
* [ ] **Code Quality**. This is much improved from the previous pass. CentreForHydrology/WDPM#18 and CentreForHydrology/WDPM#70 identify a few further improvements that I think would improve readability and modularity.

**Comment by kbarnhart** [comment]
@r-barnes thank you for taking another pass at this review. @KevinShook let me know if you have questions for me about any of these comments. It looks like you've started to address them.

**Comment by KevinShook** [comment]
Thanks. Yes, we've already fixed a couple of the issues and are working on the code quality right now.

**Comment by kbarnhart** [comment]
@KevinShook FYI, I will be out of the office Thursday June 3-Tuesday June 8.

**Comment by kbarnhart** [comment]
@KevinShook I wanted to check in and see how you were doing with addressing remaining comments.

**Comment by KevinShook** [comment]
Thanks. I've just been in contact with the other team members. We should be submitting something shortly.

**Comment by kbarnhart** [comment]
@KevinShook thanks for the update!

**Comment by kbarnhart** [comment]
@KevinShook I wanted to check in again to see how you were doing with addressing the remaining comments.

**Comment by KevinShook** [comment]
Thanks for getting back.
We have responded to issues https://github.com/CentreForHydrology/WDPM/issues/70  and https://github.com/CentreForHydrology/WDPM/issues/69 and
https://github.com/CentreForHydrology/WDPM/issues/68 and

**Comment by kbarnhart** [comment]
@KevinShook I've looked over these and agree that you have addressed all issues except for the suggestion surrounding tests. I've provided some input on that issue. Please let me know on that issue if I can provide additional guidance. Once that issue is resolved I will ping @r-barnes to revisit his review.

**Comment by kbarnhart** [comment]
@r-barnes as best as I can tell @KevinShook has addressed the remaining issues you have raised. As you are able, please revisit your review.

**Comment by kbarnhart** [comment]
@r-barnes a friendly reminder to revisit this review as you are able.

**Comment by kbarnhart** [comment]
@r-barnes another friendly reminder to revisit this review as you are able.

**Comment by kbarnhart** [comment]
@r-barnes another friendly reminder to revisit this review as soon as you are able. If I don't hear from you in a few days I'm going to move ahead with this submission as I believe the authors have addressed your outstanding issues. Thanks!

**Comment by kbarnhart** [comment]
@KevinShook after discussion with others on the JOSS editorial board, I have decided to move forward with this submission. 

Many thanks to all three reviewers @r-barnes @awickert and @KCallaghan for your constructive reviews. I believe that the authors have now addressed the outstanding comments, and I have gone ahead and completed the remaining checklist items/closed issues. 

The checklist items I checked from @r-barnes 's checklist were:
Functionality: Have the functional claims of the software been confirmed?
Functionality documentation: Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
Automated tests: Are there automated tests or manual steps described so that the functionality of the software can be verified?
Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support
 State of the field: Do the authors describe how this software compares to other commonly-used packages?

**Comment by kbarnhart** [comment]
@whedon check references

**Comment by kbarnhart** [comment]
@whedon compile pdf

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.envsoft.2020.104850 is OK
- 10.1016/j.envsoft.2020.104850 is OK
- 10.5194/esurf-8-431-2020 is OK
- 10.5194/esurf-9-105-2021 is OK
- 10.5194/esurf-7-737-2019 is OK
- 10.1029/2020WR027984 is OK
- 10.1002/hyp.10567 is OK
- 10.1007/s10669-018-9684-7 is OK
- /10.1016/S0022-1694(98)00098-5 is OK
- 10.1111/j.1745-6584.1988.tb00794.x is OK
- 10.1002/esp.3888 is OK
- 10.1002/hyp.6787 is OK
- 10.1002/hyp.9409 is OK
- 10.1002/hyp.8381 is OK
- 10.1016/j.jhydrol.2020.125846 is OK
- 10.1002/hyp.9867 is OK
- 10.1016/j.jhydrol.2014.12.014 is OK
- 10.1016/S0022-1694(03)00175-6 is OK
- 10.3390/w11122486 is OK
- 10.1016/0022-1694(93)90275-E is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kbarnhart** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02276/joss.02276/10.21105.joss.02276.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02276/joss.02276/10.21105.joss.02276.pdf) :page_facing_up: :point_left:

**Comment by kbarnhart** [comment]
@KevinShook a few final comments from me:
1. In the paper, please fix the .bib entry for both Venema references, I think the curly braces should be removed so the author is listed as Venema, H.D. (or correct me if you think otherwise). 
2. Over the last year we have started enforcing that the papers actually contain a section called "Statement of Need". I would recommend you rename "Program Description" to "Statement of Need" and then add another section title around line 39 to demarcate the transition between the prior section and your nice comparison between this and other work. 

Once you have done both of those things, please verify that `@whedon check references` and `@whedon generate pdf` work without error (e.g., that all DOIs are resolved and the PDF successfully compiles).  At that point, please do the following: 

 - [ ] Make a tagged release of your software, and list the version tag of the archived version here.
 - [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository).
 - [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
 - [ ] Please list the DOI of the archived version here.
 
I can then move forward with accepting the submission.

I will be out of the office on Friday August 6th

**Comment by KevinShook** [comment]
Thanks very much. I'll take care of these.

**Comment by KevinShook** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.envsoft.2020.104850 is OK
- 10.1016/j.envsoft.2020.104850 is OK
- 10.5194/esurf-8-431-2020 is OK
- 10.5194/esurf-9-105-2021 is OK
- 10.5194/esurf-7-737-2019 is OK
- 10.1029/2020WR027984 is OK
- 10.1002/hyp.10567 is OK
- 10.1007/s10669-018-9684-7 is OK
- /10.1016/S0022-1694(98)00098-5 is OK
- 10.1111/j.1745-6584.1988.tb00794.x is OK
- 10.1002/esp.3888 is OK
- 10.1002/hyp.6787 is OK
- 10.1002/hyp.9409 is OK
- 10.1002/hyp.8381 is OK
- 10.1016/j.jhydrol.2020.125846 is OK
- 10.1002/hyp.9867 is OK
- 10.1016/j.jhydrol.2014.12.014 is OK
- 10.1016/S0022-1694(03)00175-6 is OK
- 10.3390/w11122486 is OK
- 10.1016/0022-1694(93)90275-E is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by KevinShook** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02276/joss.02276/10.21105.joss.02276.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02276/joss.02276/10.21105.joss.02276.pdf) :page_facing_up: :point_left:

**Comment by KevinShook** [comment]
Should I be storing the paper on Zenodo as well? As it is under a CC4 licence, I would assume that it would have to be kept separate from the program which is GPL3 - is that correct?

**Comment by kbarnhart** [comment]
The entire repo at https://github.com/CentreForHydrology/WDPM is what should be in the Zenodo archive. 

I'd expect that the .bib and the .md for the paper are covered under the repository's license, while the final publication is what would be under the CC4. I am not a license expert though. 

I think that answers the question, but if it doesn't, let me know.

**Comment by KevinShook** [comment]
Ah, thanks very much, that's a big help

**Comment by KevinShook** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02276/joss.02276/10.21105.joss.02276.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02276/joss.02276/10.21105.joss.02276.pdf) :page_facing_up: :point_left:

**Comment by KevinShook** [comment]
The tagged release (v2.0b) is now on GitHub:
https://github.com/CentreForHydrology/WDPM/releases/tag/v2.0b

**Comment by KevinShook** [comment]
The archived code is now on Zenodo.
DOI: 10.5281/zenodo.5165172

Thanks very much for all the help

**Comment by kbarnhart** [comment]
@whedon set v2.0b as version

**Comment by whedon** [comment]
OK. v2.0b is the version.

**Comment by kbarnhart** [comment]
@whedon set 10.5281/zenodo.5165172 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5165172" target="_blank">10.5281/zenodo.5165172</a> is the archive.

**Comment by kbarnhart** [comment]
@whedon recommend accept

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by kbarnhart** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2505

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2505, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.envsoft.2020.104850 is OK
- 10.1016/j.envsoft.2020.104850 is OK
- 10.5194/esurf-8-431-2020 is OK
- 10.5194/esurf-9-105-2021 is OK
- 10.5194/esurf-7-737-2019 is OK
- 10.1029/2020WR027984 is OK
- 10.1002/hyp.10567 is OK
- 10.1007/s10669-018-9684-7 is OK
- /10.1016/S0022-1694(98)00098-5 is OK
- 10.1111/j.1745-6584.1988.tb00794.x is OK
- 10.1002/esp.3888 is OK
- 10.1002/hyp.6787 is OK
- 10.1002/hyp.9409 is OK
- 10.1002/hyp.8381 is OK
- 10.1016/j.jhydrol.2020.125846 is OK
- 10.1002/hyp.9867 is OK
- 10.1016/j.jhydrol.2014.12.014 is OK
- 10.1016/S0022-1694(03)00175-6 is OK
- 10.3390/w11122486 is OK
- 10.1016/0022-1694(93)90275-E is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kbarnhart** [comment]
@KevinShook I've now recommended that this submission be accepted. One of the @openjournals/joss-eics will handle the final steps of this process. Congratulations on a valuable contribution. Many thanks to reviewers @r-barnes @awickert @KCallaghan for you thoughtful and constructive reviews.

**Comment by KevinShook** [comment]
Wonderful. Thanks so much again for shepherding us through the process, and thanks as well to the reviewers for making the paper and the program much better.

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1426074501054357509) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2510
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02276](https://doi.org/10.21105/joss.02276)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@r-barnes, @awickert, @KCallaghan – many thanks for your reviews here and to @kbarnhart for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@KevinShook – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02276/status.svg)](https://doi.org/10.21105/joss.02276)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02276">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02276/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02276/status.svg
   :target: https://doi.org/10.21105/joss.02276
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02276/status.svg)](https://doi.org/10.21105/joss.02276)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
